storage/remote/queue_manager.go

```diff
@@ -1084,66 +1169,65 @@ func (s *shards) runShard(ctx context.Context, shardID int, queue chan interface
 			s.exemplarsDroppedOnHardShutdown.Add(uint32(droppedExemplars))
 			return
 
-		case sample, ok := <-queue:
+		case batch, ok := <-queue.Chan():
```

Comment:
should we just save the result of `queue.Chan()` somewhere earlier in `runShard`?

Reply:
Done!